source/grid/grid_in.cc

```diff
@@ -3563,19 +3563,18 @@ namespace
 
     AssertThrow(input_stream, ExcIO());
     std::string line;
-    std::getline(input_stream, line);
 
-    while (!input_stream.fail() && !input_stream.eof())
+    while (std::getline(input_stream, line))
       {
+      cont:
```

Comment:
Why are you moving the label here? This changes how the loop is working because now the condition in the `while` is not checked anymore

Reply:
In all of the instances of `goto` in this function (except on line 3840) , a check is made to make sure `line` is non-empty and starts with `'*'`. At the jump of the control flow, `line` already contains a valid Abaqus command and we don't need to check the `while` condition on line 3567 (checking the `while` condition wrongfully replaces the value of `line`). I agree that we need to pay extra attention here and even try to remove need for `goto`, but this patch seems to work properly, passing the tests and fixing the newline bug.